components/TodoForm.tsx

```diff
@@ -0,0 +1,205 @@
+'use client'
+
+import { Mode, Task } from '@/app/types/tasks'
+import { useState } from 'react'
+import TaskDetails from './TaskDetails'
+
+import Image from 'next/image'
+
+//Import Svg for icons
+import calendarIcon from "@/public/assets/calendar.svg"
+import AccountIcon from "@/public/assets/profile.svg"
+import StatusIcon from "@/public/assets/status.svg"
+import TagsIcon from "@/public/assets/priceTag.svg"
+import IDIcon from "@/public/assets/id.svg"
+import SaveIcon from "@/public/assets/save.svg"
+import SendIcon from "@/public/assets/send.svg"
+import { FormEvent } from 'react'
+import { FORM_MODE } from '@/app/constants/Task'
+
+
+
+interface TodoFormProps {
+    initialData?: Task
+    onSubmit?: (data: Task) => void
+    mode: Mode;
+    onAcknowledge?: () => void
+    onClose: () => void
+}
+
+export default function TodoForm({ initialData, onSubmit, mode, onAcknowledge, onClose }: TodoFormProps) {
+    const [formData, setFormData] = useState<Task>({
+        id: initialData?.id || '',
+        title: initialData?.title || '',
+        description: initialData?.description || '',
+        dueDate: initialData?.dueDate || '',
+        assignee: initialData?.assignee || '',
+        tags: initialData?.tags || '',
+        taskId: initialData?.taskId || '',
+        status: initialData?.status || 'Todo',
+    })
```

Comment:
- let's create a default config for `formData` and use it here.

```suggestion
    const DEFAULT_FORM_DATA = {
        id: '',
        title: '',
        description: '',
        dueDate: '',
        assignee: '',
        tags: '',
        taskId: '',
        status: 'Todo',
    }
    
    const [formData, setFormData] = useState<Task>(initialData ?? DEFAULT_FORM_DATA)
```

Reply:
- move this default object outside of the react component.